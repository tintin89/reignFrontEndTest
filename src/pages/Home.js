import React from 'react'
import '../App.css';
import Results from '../components/Results/Results';
import {useSelector} from 'react-redux';  
import Dropdown from '../components/Dropdown/Dropdown';

const mapState = (myStateApp)=>({
    posts:myStateApp.posts
})



function Home() {
    const {posts} = useSelector(mapState);
    return (   
        <>
        <Dropdown/>
        <main>               
         <Results arr={posts}/>    
      </main>
      
        </>
       
    )
}

export default Home
