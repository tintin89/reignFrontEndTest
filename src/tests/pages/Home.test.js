import React from 'react';
import  { shallow }  from "enzyme";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Home from '../../pages/Home';

const mockStore = configureStore();

describe('tests in  <Home/>',()=>{   
    const initState = {
        posts: [],        
    };
    let store = mockStore(initState);

    test('Must show correctly the component',()=>{
       
        const wrapper = shallow(
            <Provider store={store}>
             <Home/>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot()
    })

   
})