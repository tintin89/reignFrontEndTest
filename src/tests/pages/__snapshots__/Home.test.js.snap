// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in  <Home/> Must show correctly the component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Home />
</ContextProvider>
`;
